clientlibrary/config/config.go

```diff
@@ -172,6 +173,12 @@ type (
 		// StreamName is the name of Kinesis stream
 		StreamName string
 
+		// KinesisIAMRoleArn is the IAM Role to access Kinesis stream
```

Comment:
Please follow the original AWS KCL's approach by providing AwsCredentialsProvider instead.
https://github.com/awslabs/amazon-kinesis-client/blob/master/amazon-kinesis-client-multilang/src/main/java/software/amazon/kinesis/coordinator/KinesisClientLibConfiguration.java#L211

In aws-go-sdk, there is a Provider interface and it will be used to retrieve AWS credentials:
https://github.com/aws/aws-sdk-go/blob/master/aws/credentials/credentials.go#L202

Are you trying to access Kinesis/DynamoDB inside different AWS account using cross-account-trust?

I think you can do this by either creating a non-privileged IAM user in target account and use it's credentials. Or, modify the instance profile of your EC2 instance by allowing the instance to assume that role. 
https://docs.aws.amazon.com/IAM/latest/UserGuide/tutorial_cross-account-with-roles.html

If this is the case, I don't think the code change is needed.

Reply:
This is just to enable role based access since we operate using KIAM in kubernetes (which uses temporary credentials by assuming the roles). Initially I did have separate roles one each of Kinesis/DynamoDB etc, but then on updating my PR I decided to use a single role that encompasses policies for all the resources involved.

Are you saying, that I have a separate role which can have policies specific to the resources (like Kinesis, Dynamodb. Cloudwatch ) instead of a single role that can have policies from 
 all these resources.

Also, the use case is not necessarily regarding access to a cross account resource

Reply:
Please add 3 providers instead of adding the RoleARN. This will be consistent with original AWS KCL.
https://github.com/aws/aws-sdk-go/blob/master/aws/credentials/credentials.go#L86

If you just use temporary credentials by assuming the roles, you don't need to make any code change at all. Just create an inline policy to your instance profile and add permission to the resources (Kinesis/DynamoDB/CloudWatch).

Reply:
or do you mean to say use 
ProfileAssumeRoleCredentialsProvider which is an implementation of  AwsCredentialsProvider

Reply:
ProfileAssumeRoleCredentialsProvider works automatically if the code is running on the EC2 instance. You need to configure the instance profile of the EC2 instance. 
https://docs.amazonaws.cn/en_us/autoscaling/ec2/userguide/us-iam-role.html

Once instance profile has been configured correctly, there isn't any code change needed. But, I do see the value of adding Provider interface. Again, for your case, no code change is needed.

Reply:
Thanks, will look at providing 3 providers, but then based on some config the role arn will have to be provided to assume the role.

Reply:
Configuring the instance profile of the EC2 instance is not an option for us as we run on provisioned kubernetes cluster shared by different applications. So our apps could share the same node as some other app at one time and on a different occasion, our pods could be running on different nodes of the cluster. Again, the cluster is shared.

Also, looks like using provider interface  in go is not uniform like in java across different providers. The interfaces seems to be different, if you look at stscreds package, the credentials package and all other packages under credentials. Correct me if I am wrong...